lib/logger.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const pino = require('pino')
+
+const startTime = Symbol('startTime')
+
+function createLogger (opts, stream) {
+  stream = stream || opts.stream
+  delete opts.stream
+
+  var prevLogger = opts.logger
+  var prevGenReqId = opts.genReqId
+  var logger = null
+
+  if (prevLogger) {
+    opts.logger = undefined
+    opts.genReqId = undefined
+    logger = prevLogger.child(opts)
+    opts.logger = prevLogger
+    opts.genReqId = prevGenReqId
+  } else {
+    logger = pino(opts, stream)
+  }
+
+  return logger
+}
+
+function reqIdGenFactory (func) {
+  if (typeof func === 'function') return func
+  var maxInt = 2147483647
+  var nextReqId = 0
+  return function genReqId (req) {
```

Comment:
Why `req` parameter is needed here?
The generation algorithm doesn't need it

Reply:
@delvedor here ?

Reply:
Sorry, I missed it.

Reply:
`req`  is used in the line after this, in case there is an id already set, e.g. `req.id`. This is from pino-http and can be removed.